frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { compose } from 'redux';
+
+import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import DetailItem from '@console/shared/src/components/dashboard/details-card/DetailItem';
+import DetailsBody from '@console/shared/src/components/dashboard/details-card/DetailsBody';
+import { usePrometheusQueries } from '@console/shared/src/components/dashboard/utilization-card/prometheus-hook';
+
+import { BlockPoolDashboardContext } from './block-pool-dashboard-context';
+import { getPoolQuery, StorageDashboardQuery } from '../../../queries/ceph-queries';
+
+const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
+
+export const CompressionDetailsCard: React.FC = () => {
+  const { t } = useTranslation();
+  const { obj } = React.useContext(BlockPoolDashboardContext);
+  const { compressionMode } = obj.spec;
```

Comment:
It is better you use: const compressionMode  = obj.spec?.compressionMode; it will fail if the pool is created from CLI and if the user immediately opens the pool dashboard page after pool creation.

Reply:
Done.